src/main/java/uk/gov/pay/webhooks/util/MicrosecondPrecisionInstantDeserializer.java

```diff
@@ -0,0 +1,25 @@
+package uk.gov.pay.webhooks.util;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+
+public class MicrosecondPrecisionInstantDeserializer extends JsonDeserializer<Instant> {
+
+    public static final DateTimeFormatter MICROSECOND_FORMATTER =
+            MicrosecondPrecisionInstantSerializer.MICROSECOND_FORMATTER;
+    
+
+    @Override
+    public Instant deserialize(JsonParser p, DeserializationContext ctxt)  {
+        try {
+            return Instant.from(MICROSECOND_FORMATTER.parse(p.getText()));
```

Comment:
I think we can get away without needing to rely on a `DateTimeFormatter` here: we can just use `Instant.parse(p.getText())`. This will will be slightly more liberal in terms of what it accepts but since everything it‘s parsing will be created by us, I think that’s OK. Without the `DateTimeFormatter` this class could then become `InstantDeserializer`, which will be more reusable.